### PR TITLE
Issue #19: update generator to generate patch file by 'git diff' commands

### PR DESCRIPTION
**Pull request by HuGanghui**
Issue #19: update generator to generate patch file by 'git diff' commands

### CODE REVIEW

**Comment by HuGanghui** [comment]
@timurt please review and give your advice, is this ok?

File: src/main/java/com/github/checkstyle/generatepatchfile/GeneratePatchFile.java [hunk 1]
```diff
@@ -277,6 +263,25 @@ private void generateDiffPatchWithGitCommand(int headNum, String patchFormat) th
...
+        final List<String> showCmds = new ArrayList<>();
+        showCmds.add("sh");
+        showCmds.add("-c");
+        showCmds.add(command);
+        final Process process = new ProcessBuilder()
+                .directory(new File(repoPath))
+                .command(showCmds)
+                .start();
+        final int code = process.waitFor();
+        stateException(code, "an error occurred when running" + command);
```
Comment (baratali) [inline src/main/java/com/github/checkstyle/generatepatchfile/GeneratePatchFile.java:276]: please put a space at the end of the string:
`"an error occurred when running "`

File: src/main/java/com/github/checkstyle/generatepatchfile/GeneratePatchFile.java [hunk 2]
```diff
@@ -277,6 +263,25 @@ private void generateDiffPatchWithGitCommand(int headNum, String patchFormat) th
         }
     }
 
+    private void runShellCommand(String command) throws Exception {
+        final List<String> showCmds = new ArrayList<>();
+        showCmds.add("sh");
```
Comment (baratali) [inline src/main/java/com/github/checkstyle/generatepatchfile/GeneratePatchFile.java:268]: please remove 'show' word from names

File: src/main/java/com/github/checkstyle/generatepatchfile/GeneratePatchFile.java [hunk 3]
```diff
@@ -227,31 +222,22 @@ private void generateTwoCommitDiffPatch(RevCommit commitOld, RevCommit commitNew
...
                     .directory(new File(repoPath))
                     .command(GIT, "format-patch", "-1")
                     .inheritIO()
                     .start();
             final int code = process.waitFor();
-            if (code != 0) {
-                throw new IllegalStateException(
-                        "an error occurred when running git format-patch -1");
-            }
+            stateException(code, "an error occurred when running git format-patch -1");
```
Comment (baratali) [inline src/main/java/com/github/checkstyle/generatepatchfile/GeneratePatchFile.java:237]: can you use `runShellCommand("git format-patch -1")` here?

File: src/main/java/com/github/checkstyle/generatepatchfile/GeneratePatchFile.java [hunk 4]
```diff
@@ -172,7 +167,7 @@ public void generatePatch(int runPatchNum) throws Exception {
      * @throws Exception exception
      */
     public void generatePatchWithGitCommand(int runPatchNum, String patchFormat) throws Exception {
-        for (int i = 1; i < runPatchNum; i++) {
+        for (int i = 1; i < runPatchNum + 1; i++) {
```
Comment (baratali) [inline src/main/java/com/github/checkstyle/generatepatchfile/GeneratePatchFile.java:170]: why did you increase amount of patches? please move to a new commit if it's not related to new `diff` patch format

File: src/main/java/com/github/checkstyle/generatepatchfile/GeneratePatchFile.java (hunk 1 above)
Comment (HuGanghui) [reply src/main/java/com/github/checkstyle/generatepatchfile/GeneratePatchFile.java]: done

File: src/main/java/com/github/checkstyle/generatepatchfile/GeneratePatchFile.java (hunk 2 above)
Comment (HuGanghui) [reply src/main/java/com/github/checkstyle/generatepatchfile/GeneratePatchFile.java]: done

File: src/main/java/com/github/checkstyle/generatepatchfile/GeneratePatchFile.java (hunk 3 above)
Comment (HuGanghui) [reply src/main/java/com/github/checkstyle/generatepatchfile/GeneratePatchFile.java]: done

File: src/main/java/com/github/checkstyle/generatepatchfile/GeneratePatchFile.java (hunk 4 above)
Comment (HuGanghui) [reply src/main/java/com/github/checkstyle/generatepatchfile/GeneratePatchFile.java]: done

File: src/main/java/com/github/checkstyle/generatepatchfile/GeneratePatchFile.java [hunk 5]
```diff
@@ -281,6 +261,26 @@ private void generateDiffPatchWithGitCommand(int headNum, String patchFormat) th
...
+        cmds.add("sh");
+        cmds.add("-c");
+        cmds.add(command);
+        final Process process = new ProcessBuilder()
+                .directory(new File(repoPath))
+                .command(cmds)
+                .inheritIO()
+                .start();
+        final int code = process.waitFor();
+        stateException(code, "an error occurred when running " + command);
```
Comment (timurt) [inline src/main/java/com/github/checkstyle/generatepatchfile/GeneratePatchFile.java:275]: why exception not thrown right here? do you really separate method?

Comment (HuGanghui) [reply src/main/java/com/github/checkstyle/generatepatchfile/GeneratePatchFile.java]: done, yes it is not necessary to separate method.

**Pull request merged**